<?xml version="1.0"?>

<launch>

    <arg name="unpause_timeout" default="3"/>
    <!-- <arg name="world" default="worlds/gas_infrastructure.world"/> -->
    <arg name="world" default="$(find remaro_scenarios)/worlds/gas_infrastructure_random.world"/>

    <!-- Start the launch file with record:=false if it is not being used by the optimizer -->
    <arg name="record" default="false"/>
    <arg name="bag_filename" default="helical_inspection_recording.bag"/>
    
    <!-- Setup for a timeout for the simulation run -->
    <arg name="set_timeout" default="false"/>
    <arg name="timeout" default="400"/>

    <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name"   value="$(arg world)"/>
        <arg name="paused"       value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui"          value="true"/>
        <arg name="headless"     value="false"/>
        <arg name="debug"        value="false"/>
        <arg name="verbose"      value="false"/>
    </include>

    <!-- Vehicle's initial position -->
    <arg name="x" default="20"/>
    <arg name="y" default="0"/>
    <arg name="z" default="-13"/>
    <arg name="yaw" default="0"/>


    <!-- Start spawning vehicle and take control -->
    <include file="$(find bluerov2_gazebo)/launch/start_pid_controller_demo.launch">
        <arg name="teleop_on" value="true"/>
        <arg name="x" default="$(arg x)"/>
        <arg name="y" default="$(arg y)"/>
        <arg name="z" default="$(arg z)"/>
        <arg name="yaw" default="$(arg yaw)"/>
    </include> 
    
    <!-- Start executing helical trajectory -->
    <include file="$(find uuv_control_utils)/launch/start_helical_trajectory.launch">
        <arg name="uuv_name" value="bluerov2"/>
        <arg name="radius" value="4"/>
        <arg name="delta_z" value="-3"/>
        <arg name="n_points" value="50"/>
        <arg name="n_turns" value="4"/>
        <arg name="center_x" default="20"/>
        <arg name="center_y" default="7"/>
        <arg name="center_z" default="-13"/>
        <arg name="heading_offset" default="0"/>
        <arg name="duration" default="0"/>
        <arg name="max_forward_speed" value="0.9"/>
    </include>

--> 

     <!-- <include file="$(find remaro_auv_description)/launch/upload.launch">
        <arg name="x" default="-1"/>
        <arg name="y" default="-20"/>
        <arg name="z" default="-17"/>
        <arg name="roll" default="0.0"/>
        <arg name="pitch" default="0.0"/>
        <arg name="yaw" default="0.0"/>
    </include>  -->


    
      <!-- <node pkg="rosbag" type="record" name="_recording"
        args="record -O $(arg bag_filename)
        clock
        /rexrov/ground_truth_to_tf_rexrov/pose/pose/position/x
        /rexrov/ground_truth_to_tf_rexrov/pose/pose/position/y
        /rexrov/ground_truth_to_tf_rexrov/pose/pose/position/z
        "
        if="$(arg record)"/> -->

  <!--  <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find bluerov2_control)/rviz/bluerov2_control.rviz"/> -->
</launch>
